### DISCUSSION TITLE
Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS   HDF5_HL_LIBRARIES C HL) (found version "")

### DISCUSSION THREAD

**Discussion by nancysemwal** [idaholab/moose#29043] [Q&A Getting Started]

**Comment by GiudGiud** [comment]
Hello

for HDF5, if you dont have it, it should be installed by PETSc. Then you can set HDF5_DIR to the folder containing the hdf5 installation. 

If you do have it, then set HDF5_DIR

**Reply by GiudGiud** [reply]
for installing Cardinal with conda, no idea but I dont think it's possible?

**Reply by milljm** [reply]
1. Entry point fixes are more than likely going to require the reinstallation of Conda. With Conda exhibiting this error, I feel we can't trust Conda to be doing the right things.

2. You have a system installation of MPI and an MPI installation from Conda (by using our packages):
```pre
-L/usr/lib/x86_64-linux-gnu/openmpi/lib/fortran/gfortran
/home/nancy/miniconda3/envs/moose/bin/mpicc
```
Many odd occurrences will crop up by allowing both to exist.

3. Are you building on a thumbstick, or external drive? I am referring to the `/mnt` path for Cardinal. While this is fine, it may explain why missing libraries can exist; Are you building this on one machine and bringing it to another? FYI, that won't work well. Not unless both machines are identically built.


The issues you are running into seems to me, being caused by a not-sane environment. We need to clean this up. Because you have OpenMPI installed on the system, I would refrain from using any of our Conda packages, and just stick to using `apt-get` to install what you need.

It looks like the first run of your `./diagnostics.sh` was only missing `F77 F90` variables:
```bash
# get out of the moose environment
conda deactivate
conda activate base

# set variables that the system installed version of OpenMPI is not for some reason:
export F77=mpif77 F90=mpif90
./diagnostics.sh
```

Because you introduced two completely different compiler stacks into the mix, the best thing to do from here (`./diagnostics.sh` is passing and you are not loading the moose environment), is start completely over. The entire Cardinal build must be built using the same compiler.

**Reply by milljm** [reply]
Going back and reading the other post yields more concerns:

Before, when you had the base environment loading, there was a different Python version being made available, and further more, was not a Conda variant:

```pre
CONDA_PREFIX=/home/nancy/miniconda3
CONDA_PROMPT_MODIFIER=(base)
CONDA_PYTHON_EXE=/home/nancy/miniconda3/bin/python

<trimmed>

Python Sanity Checks

/usr/bin/env python3 --version; (reporting as: Python 3.12.3) matches
which python3 python;

/usr/bin/python3 --version; == Python 3.12.3
WARNING: python; does not exist
```
_looks like you were using system supplied Python back then_

But now its:
```pre
CONDA_PREFIX=/home/nancy/miniconda3
CONDA_PROMPT_MODIFIER=(base) 
CONDA_PYTHON_EXE=/home/nancy/miniconda3/bin/python

<trimmed>

Python Sanity Checks

/usr/bin/env python3 --version; (reporting as: Python 3.12.7) matches
which python3 python;

/home/nancy/miniconda3/bin/python3 --version; == Python 3.12.7
/home/nancy/miniconda3/bin/python --version; == Python 3.12.7
```
_Conda supplied Python_
Something changed between that build, and your current one. Just listing more oddities...

While not a cause for failure, it is just _odd_ is all I am pointing out.

**Reply by nancysemwal** [reply]
Yes, you're right about the Python thing.

Initially I was using the system supplied Python but then I changed my `$PATH` to pick up the `/home/miniconda3/bin/python` instead. 

I am indeed having a lot of inconsistencies and oddities in my setup and trying to making Cardinal work with them is causing the complexities to creep in. My reason for doing so was to not lose the progress made while building `PETSc, libmesh & wasp` because it took a loooong time. But I guess there's no other way.

**Reply by nancysemwal** [reply]
> Are you building on a thumbstick, or external drive? I am referring to the /mnt path for Cardinal. While this is fine, it may explain why missing libraries can exist; Are you building this on one machine and bringing it to another? FYI, that won't work well. Not unless both machines are identically built.

No, I'm not building on an external drive. The `/mnt` path is my `/dev/sdb4`, a 200 GB partition that I'd created from my 1 TB HDD (but not external, internal drive only, apart from a 250 GB SDD where my `/home` resides.)
